catchup/service.go

```diff
@@ -710,6 +731,28 @@ func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.Asy
 				return
 			default:
 			}
+			if err == errNoBlockForRound {
+				// If a peer does not have the block after few attempts it probably has not persisted the block yet.
+				// Give it some time to persist the block and try again.
+				// Quit if the the same peer still errs after sleeping 2*config.Protocol.SmallLambda seconds in total.
+				if info, ok := peerErrors[peer]; ok {
```

Comment:
Similar think this should have test coverage - faking it may not be trivial though.

Reply:
I removed the preliminary quitting logic and made it simpler.